### PR TITLE
Updates to OTel-Arrow v0.24.0 deps

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** 
This is a copy of the otel-arrow repository exporter and receiver components at
https://github.com/open-telemetry/otel-arrow/commit/dd6e2249a180a51433a93751680fa67dc9c6432e

Updates both go.mods to OTel-Arrow v0.24.0 dependencies, the current release.

Applies repo-specific `make gci` logic.

**Link to tracking Issue:** #26491 

**Testing:** Passing

**Documentation:** n/a

### CODE REVIEW

**Comment by jmacd** [comment]
(The test failures look unrelated.)

**Review by codeboten — COMMENTED** [review]
Right, they're related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33520

**Review by moh-osman3 — APPROVED** [review]
LGTM

File: receiver/otelarrowreceiver/config.go [hunk 1]
```diff
@@ -24,6 +24,16 @@ type ArrowConfig struct {
 	// passing through, they will see ResourceExhausted errors.
 	MemoryLimitMiB uint64 `mapstructure:"memory_limit_mib"`
 
+	// AdmissionLimitMiB limits the number of requests that are received by the stream based on
+	// request size information available. Request size is used to control how much traffic we admit
+	// for processing, but does not control how much memory is used during request processing.
+	AdmissionLimitMiB uint64 `mapstructure:"admission_limit_mib"`
```
Comment (codeboten) [inline receiver/otelarrowreceiver/config.go:30]: Should these options be added to the README for the component?

**Review by codeboten — COMMENTED** [review]
Thanks @jmacd. Will there be much more work on the receiver/exporter that requires it to be done in multiple repositories or could most of the work be done in this repo?

Reason I'm asking is that it's difficult to catch up on all the changes in a single PR review. And I'm guessing doing the work in multiple places sucks too :)

**Review by jmacd — COMMENTED** [review]
@codeboten Thanks! Re: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33518#pullrequestreview-2114252010, yes ... yes.

I filed https://github.com/open-telemetry/otel-arrow/issues/225 to record this intention. We will remove the exporter and receiver components from that repository as soon as they are merged in this PR.

**Review by codeboten — APPROVED** [review]

**Pull request merged**